update_twiki.py

```diff
@@ -0,0 +1,109 @@
+"""
+Twiki module: Creates a database for missing samples(too be expanded and completed)
+"""
+import sys
+import sqlite3
+import logging
+import numpy as np
+sys.path.append('/afs/cern.ch/cms/PPD/PdmV/tools/McM/')
+from rest import McM
+#XSDB wrapper
+from request_wrapper import RequestWrapper
+#XSDB pycurl requester instance
+xsdb_request = RequestWrapper()
+# McM instance
+mcm = McM(dev=False, debug=False, cookie='cookie.txt')
+# Logger
+logging.basicConfig(format='[%(asctime)s][%(levelname)s] %(message)s', level=logging.INFO)
+logger = logging.getLogger()
+
+def create_table():
+    conn = sqlite3.connect('twiki.db')
+    c = conn.cursor()
+    # Create table if it does not exist
+    c.execute('''CREATE TABLE IF NOT EXISTS twiki_samples
+                 (prepid text NOT NULL,
+                  dataset text NOT NULL,
+                  extension text,
+                  total_events integer NOT NULL,
+                  campaign text NOT NULL,
+                  resp_group text NOT NULL,
+                  cross_section float NOT NULL,
+                  fraction_negative_weight float NOT NULL,
+                  target_num_events real NOT NULL   )''')
+
+    # Clear the table
+    c.execute('DELETE FROM `twiki_samples`')
+    conn.commit()
+    conn.close()
+def insert_update(twiki_request, campaign, cross_section, frac_neg_wgts, target_num_events):
+    conn = sqlite3.connect('twiki.db')
+    c = conn.cursor()
+    logger.info('Inserting %s (%s) %s %s %s', twiki_request['dataset_name'], twiki_request['prepid'], cross_section, frac_neg_wgts, target_num_events)
+    c.execute('INSERT INTO twiki_samples VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)',
+              [twiki_request['prepid'],
+               twiki_request['dataset_name'],
+               twiki_request['extension'],
+               twiki_request['total_events'],
+               campaign,
+               twiki_request['prepid'].split('-')[0],
+               cross_section,
+               frac_neg_wgts,
+               target_num_events])
+
+    conn.commit()
+    conn.close()
+def operations():
+    query_ul18 = 'member_of_campaign=RunIISummer19UL18MiniAOD'
+    query_ul16 = 'member_of_campaign=RunIISummer19UL16MiniAOD'
+    query_ul17 = 'member_of_campaign=RunIISummer19UL17MiniAOD'
+    requests_ul17 = mcm.get('requests', query=query_ul17)
+    requests_ul18 = mcm.get('requests', query=query_ul18)
+    requests_ul16 = mcm.get('requests', query=query_ul16)
+    missing_ul18 = []
+    missing_ul16 = []
+    ul17_dataset_names = [x['dataset_name'] for x in requests_ul17]
```

Comment:
This is not a mistake, and I am making these two assumptions:
  1. You don't care about order of elements in ul_1X_dataset_names
  2. You want just unique values

If yes, I would suggest to use python sets. They guarantee unique values and search "in" sets are always O(1) while lists have "in" equal to O(n).
If you use sets, then you can get rid of numpy (one less thing to depend on...) and do set subtraction:
```
ul17_dataset_names = {x['dataset_name'] for x in requests_ul17}
ul16_dataset_names = {x['dataset_name'] for x in requests_ul16}
ul18_dataset_names = {x['dataset_name'] for x in requests_ul18}

missing_ul18 = ul17_dataset_names - ul18_dataset_names
missing_ul16 = ul17_dataset_names - ul16_dataset_names
```
Note that set comprehension is {...} instead of list's [...].